iOS-411 Implement delete server data

ErnestFan opened the pull request:
**What's this do?**
Implement UI for `Delete my SimplyE data`
Remove NYPLSettingsAdvancedVC

**Why are we doing this? (w/ JIRA link if applicable)**
[iOS-411](https://jira.nypl.org/browse/IOS-411)

**How should this be tested? / Do these changes have associated tests?**
1. Log in to NYPL
2. Go to Settings -> Accounts -> NYPL
3. Enable `Sync Bookmarks`
4. Select `Delete my SimplyE data` which is under `Sync Bookmarks`
5. Press `Delete Reading Data` button on the screen that shown up
6. With internet connection, you should be back to the account detail page and an alert of deletion succeed should show up
7. Without internet connection, you should stay on the same page and a `Please try again` alert should show up

**Dependencies for merging? Releasing to production?**
N/A

**Does this include changes that require a new SimplyE/Open eBooks build for QA?**
Will bump build number

**Has the application documentation been updated for these changes?**
N/A

**Did someone actually run this code to verify it works?**
@ErnestFan

ErnestFan commented:
@ettore This is mostly done except waiting for some wordings from Marcus. Once I heard back from him, I will need your help on translating it to Italian and I will update the PR after.
Will also bump the build number once everything is ready.

ettore commented:
@ErnestFan I think this is just for SimplyE and we need to keep the advanced settings VC for open ebooks, or am I wrong?

killerpuppytails commented:
Per a Slack convo today, Miguel is fine with this being the same on Open eBooks, with the name of the app correctly subbed out.

ErnestFan commented:
I will keep it as-is then.
@ettore While working on a similar screen for iOS-413, I find that it's better to use tableview instead of the stackview that I am using now. I will update that part of code, will keep you posted once it's ready for review.

ErnestFan commented:
> I will keep it as-is then. @ettore While working on a similar screen for iOS-413, I find that it's better to use tableview instead of the stackview that I am using now. I will update that part of code, will keep you posted once it's ready for review.

Updated with tableview, still waiting for wordings though. The DeleteServerDataVC was not included in every target, so I fixed that as well.

ettore commented on Simplified/Settings/NYPLSettingsDeleteServerDataViewController.swift:
```diff
@@ -0,0 +1,171 @@
+    paragraphStyle.lineSpacing = 1.25
+    paragraphStyle.lineHeightMultiple = 1.25
+    
+    let description = NSLocalizedString("DeleteServerDataDescription",
```
if you switch to Spanish, is the English translation displayed? I'm afraid "DeleteServerDataDescription" will be displayed instead.... if so you might need to copy the English wording into Spanish and German, sadly.

ettore commented on Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift:
```diff
@@ -0,0 +1,113 @@
+  
+  // MARK: - Life Cycle
+  
+  init(with message: String) {
```
in initializers "with" is typically not used by convention. You can just do `init(message: String)`

ettore commented on Simplified/Utilities/UI/NYPLActivityIndicatorMessageViewController.swift:
```diff
@@ -0,0 +1,113 @@
+  
+  // MARK: - Setup
+  
+  func setupUI() {
```
can this be private?

ErnestFan commented on Simplified/Settings/NYPLSettingsAccountDetailViewController.m:
```diff
@@ -1056,18 +1061,38 @@ - (UITableViewCell *)tableView:(__attribute__((unused)) UITableView *)tableView
+      
+      UIImageView *imageView = [self rightArrowImageView];
+      imageView.tintColor = NYPLConfiguration.deleteActionColor;
+      cell.accessoryView = imageView;
```
The UI requires a red arrow for the delete actions, tintColor does not work on `UITableViewCellAccessoryDisclosureIndicator`

ettore commented on Simplified/Settings/NYPLSettingsAccountDetailViewController.m:
```diff
@@ -1056,18 +1064,43 @@ - (UITableViewCell *)tableView:(__attribute__((unused)) UITableView *)tableView
+#ifdef SIMPLYE
+      cell.textLabel.text = NSLocalizedString(@"Delete my SimplyE Data", nil);
+#else
+      cell.textLabel.text = NSLocalizedString(@"Delete my OpenEBooks Data", nil);
```
I think the official spelling is "Open eBooks"
